File: Neetcode150/1.two-sum.java
import java.lang.reflect.Array;
import java.util.HashMap;

/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (hm.containsKey(nums[i])) {
                return new int[] { hm.get(nums[i]), i };
            }
            hm.put(target - nums[i], i);
        }
        return nullvalo;
    }
}
// @lc code=end

File: Neetcode150/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}
        for i in range(0, len(nums)):
            if nums[i] in mp:
                return [mp.get(nums[i]), i]
            mp[target-nums[i]] = i
            # @lc code=end
